.github/scripts/implement_agent.py

```
@@ -192,66 +190,83 @@ def load_context() -> str:
+def pick_model(client: OpenAI) -> str:
+    for model, label in CANDIDATE_MODELS:
+        print(f"[model] Trying {label} ({model})")
+        try:
+            client.chat.completions.create(
+                model=model,
+                max_tokens=8,
+                messages=[{"role": "user", "content": "hi"}],
+            )
+            print(f"[model] Using {model}")
+            return model
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)